File: hsg9-tinh/2023-hatinh-ts10/bnum/bnum.cpp
#include <iostream>

#define inputFile "bnum4.inp"
#define outputFile "bnum4.out"

using namespace std;

string n;
int prime;
int sum_digit; // tổng các chữ số

void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(inputFile, "r", stdin);
    cin >> n;
}


// Hàm kiểm tra số nguyên tố
bool isPrime(int n)
{
    if (n < 2) return false;
    if (n < 4) return true;
    if (n % 2 == 0) return false;
    if (n % 3 == 0) return false;

    // Kiểm tra n có chia hết cho 5, 7, 11, 13, 17, 19, v.v...
    for (int i = 5; i * i <= n; i += 6)
    {
        if (n % i == 0) return false;
        if (n % (i + 2) == 0) return false;
    }

    return true;
}


void process()
{
    sum_digit = 0;
    int digit;

    // Duyệt từng chữ số trong chuỗi n
    for (char c : n)
    {
        // Chuyển đổi chữ số thành số
        digit = c - '0';

        // Tính tổng bình phương
        sum_digit += digit * digit;
    }

    if (isPrime(sum_digit))
    {
        prime = 1;
    }
    else
    {
        prime = -1;
    }
}


void output()
{
    freopen(outputFile, "w", stdout);
    cout << prime << '\n' << sum_digit;
}


int main()
{
    input();
    process();
    output();

    return 0;
}

File: hsg9-tinh/2023-hatinh-ts10/bnum/bnum.py
import os
import sys

input_file = os.path.join(sys.path[0], 'bnum4.inp')
output_file = os.path.join(sys.path[0], 'bnum4.out')

n = ''
prime = False
sum_digit = 0


def input_data():
    global n
    
    with open(input_file, 'r') as f:
        n = f.readline().strip()


# Hàm kiểm tra số nguyên tố
def is_prime(x):
    if x < 2:
        return False
    if x < 4:
        return True
    if x % 2 == 0 :
        return False
    if x % 3 == 0:
        return False
    for i in range(5, int(x**0.5) + 1, 6):
        if n % i == 0:
            return False
        if n % (i + 2) == 0:
            return False

    return True


def process():
    global n, prime, sum_digit

    # Duyệt từng chữ số trong chuỗi n
    # chuyển đổi và tính tổng bình phương
    sum_digit = sum((int(c) ** 2) for c in n)
    
    prime = 1 if is_prime(sum_digit) else -1


def output(prime: int, sum_digit: int):
    with open(output_file, 'w') as f:
        f.write(f'{prime}\n{sum_digit}')


if __name__ == '__main__':
    input_data()
    process()
    output(prime, sum_digit)